src/transports/quic.rs

```diff
@@ -0,0 +1,180 @@
+                    poll.poll(&mut events, None)
+                        .expect(&format!("Can't poll QUIC transport of address {}", address));
+
+                    // Process each event.
+                    for event in events.iter() {
+                        match event.token() {
+                            NEW_CONNECTION_SERVER => {
+                                //TODO: Error handling
+                                let (_num_recv, from_addr) = socket.recv_from(&mut buf).unwrap();
+                                let mut config =
+                                    quiche::Config::new(quiche::PROTOCOL_VERSION).unwrap();
+                                accept(
+                                    &ConnectionId::from_ref(&buf[..PUBLIC_KEY_SIZE_BYTES]),
+                                    None,
+                                    address,
+                                    from_addr,
+                                    &mut config,
+                                )
+                                .unwrap();
+                                let mut peer_db = peer_db.write();
+                                if peer_db.nb_in_connections < peer_db.config.max_in_connections {
+                                    peer_db.nb_in_connections += 1;
+                                    let peer = Peer::new(Endpoint {});
+                                    peer_db.peers.push(peer);
+                                }
+                            }
+                            STOP_LISTENER => {
+                                return;
+                            }
+                            _ => {}
+                        }
+                    }
+                }
+            }
+        });
+        self.listeners.insert(address, (waker, listener_handle));
+        Ok(())
+    }
+
+    fn try_connect(
+        &mut self,
+        address: SocketAddr,
+        _timeout: Duration,
+        config: &Self::OutConnectionConfig,
+    ) -> Result<(), PeerNetError> {
+        //TODO: Use timeout
+        let config = config.clone();
+        std::thread::spawn({
+            let peer_db = self.peer_db.clone();
+            let wg = self.out_connection_attempts.clone();
+            move || {
+                let mut out = [0; PUBLIC_KEY_SIZE_BYTES + SIGNATURE_SIZE_BYTES];
+                out[..PUBLIC_KEY_SIZE_BYTES].copy_from_slice(&config.identity.to_bytes());
+                //TODO: Add a signature of the public key
+                let socket = UdpSocket::bind(config.local_addr)
+                    .expect(&format!("Can't bind TCP transport to address {}", address));
+                println!("Connecting to {}", address);
+                //TODO: Use configs for quiche passed from config object.
+                let mut quiche_config = quiche::Config::new(quiche::PROTOCOL_VERSION).unwrap();
+                quiche_config.verify_peer(false);
+                quiche_config
+                    .set_application_protos(&[b"massa/1.0"])
```

Comment:
Maybe add a const or make it configurable?

Reply:
Added a todo.